src/Trigger/RabbitMQTriggerAttributeBindingProvider.cs

```diff
@@ -39,13 +39,33 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
                 return Task.FromResult<ITriggerBinding>(null);
             }
 
-            string queueName = Resolve(attribute.QueueName);
+            string connectionString = Resolve(attribute.ConnectionStringSetting);
+
+            string queueName = Resolve(attribute.QueueName) ?? throw new InvalidOperationException("RabbitMQ queue name is missing");
 
             string hostName = Resolve(attribute.HostName);
 
+            string userName = Resolve(attribute.UserName);
+
+            string password = Resolve(attribute.Password);
+
+            int port = attribute.Port;
+
             ushort batchNumber = attribute.BatchNumber;
 
-            IRabbitMQService service = _provider.GetService(hostName, queueName);
+            if (string.IsNullOrEmpty(connectionString) && !string.IsNullOrEmpty(hostName) && ((string.IsNullOrEmpty(userName) && hostName != "localhost") ||

```

Comment:
defaulting to localhost if hostName is not specified will make this check simpler. Also, move this the condition check to Utility and add Unit tests